cmd/kcp-catalog/bind/catalogentry/cmd.go

```diff
@@ -0,0 +1,207 @@
+
+import (
+	"context"
+	"time"
+
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+
+	catalogv1alpha1 "github.com/kcp-dev/catalog/api/v1alpha1"
+	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	pluginhelpers "github.com/kcp-dev/kcp/pkg/cliplugins/helpers"
+	"github.com/kcp-dev/logicalcluster/v2"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+
+	kcpclient "github.com/kcp-dev/kcp/pkg/client/clientset/versioned"
+	"github.com/kcp-dev/kcp/pkg/cliplugins/base"
+	"k8s.io/client-go/rest"
+)
+
+// BindOptions contains the options for creating APIBindings for CE
+type BindOptions struct {
+	*base.Options
+	// WorkspacePath refers to the absolute reference to the Catalog Entry workspace
+	// to which the user wants to bind.
+	workspacePath string
+	// Name of the CE.
+	catalogentryName string
+	// APIExportRef is the argument accepted by the command. It contains the
+	// reference to where APIExport exists. For ex: <absolute_ref_to_workspace>:<apiexport>.
+	CatalogEntryRef string
+}
+
+// NewBindOptions returns new BindOptions.
+func NewBindOptions(streams genericclioptions.IOStreams) *BindOptions {
+	return &BindOptions{
+		Options: base.NewOptions(streams),
+	}
+}
+
+// BindFlags binds fields to cmd's flagset.
+func (b *BindOptions) BindFlags(cmd *cobra.Command) {
+	b.Options.BindFlags(cmd)
+}
+
+// Complete ensures all fields are initialized.
+func (b *BindOptions) Complete(args []string) error {
+	if err := b.Options.Complete(); err != nil {
+		return err
+	}
+
+	if len(args) > 0 {
+		b.CatalogEntryRef = args[0]
+	}
+	return nil
+}
+
+// Validate validates the BindOptions are complete and usable.
+func (b *BindOptions) Validate() error {
+	if b.CatalogEntryRef == "" {
+		return errors.New("`root:ws:catalogentry_object` reference to bind is required as an argument")
+	}
+
+	if !strings.HasPrefix(b.CatalogEntryRef, "root") {
+		return fmt.Errorf("fully qualified reference to workspace where Catalog Entry exists is required. The format is `root:<ws>:<catalogentry>`")
+	}
+
+	return b.Options.Validate()
+}
+
+// Run creates an apibinding for the user.
+func (b *BindOptions) Run(ctx context.Context) error {
+	config, err := b.ClientConfig.ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	kcpclient, err := newKCPClusterClient(config)
+	if err != nil {
+		return err
+	}
+
+	_, currentClusterName, err := pluginhelpers.ParseClusterURL(config.Host)
+	if err != nil {
+		return err
+	}
+
+	// get the entry refered in the command.
+	path, entryName := logicalcluster.New(b.CatalogEntryRef).Split()
+
+	client, err := newCatalogClient(config)
+	if err != nil {
+		return err
+	}
+
+	// TODO: figure out if this usage of controller-runtime client can be used.
```

Comment:
Is the question if we can use controller-runtime at all?

Reply:
My bad, I didn't push the latest changes. Its all updated to using controller-runtime clients now.